x/airdrop/types/proposals.go

```diff
@@ -0,0 +1,48 @@
+package types
+
+import (
+	fmt "fmt"
+	"strings"
+
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+)
+
+const (
+	ProposalTypeRegisterZoneDrop = "RegisterZoneDrop"
+)
+
+var _ govtypes.Content = &RegisterZoneDropProposal{}
+
+func (m RegisterZoneDropProposal) GetDescription() string { return m.Description }
+func (m RegisterZoneDropProposal) GetTitle() string       { return m.Title }
+func (m RegisterZoneDropProposal) ProposalRoute() string  { return RouterKey }
+func (m RegisterZoneDropProposal) ProposalType() string   { return ProposalTypeRegisterZoneDrop }
+
+// ValidateBasic runs basic stateless validity checks
+func (m RegisterZoneDropProposal) ValidateBasic() error {
+	err := govtypes.ValidateAbstract(m)
+	if err != nil {
+		return err
+	}
+
+	// validate ZoneDrop
+
+	// validate ClaimRecords
+
+	return nil
+}
+
+// String implements the Stringer interface.
+func (m RegisterZoneDropProposal) String() string {
+	var b strings.Builder
+
+	b.WriteString("Airdrop - ZoneDrop Registration Proposal:\n")
+	b.WriteString(fmt.Sprintf("\tTitle:       %s\n", m.Title))
+	b.WriteString(fmt.Sprintf("\tDescription: %s\n", m.Description))
+	b.WriteString("\tZoneDrop:\n")
+	b.WriteString(fmt.Sprintf("\n%v\n\n", m.ZoneDrop))
+	b.WriteString("\tClaimRecords:\n")
+	b.WriteString(fmt.Sprintf("\n%v\n", m.ClaimRecords))
```

Comment:
Not sure how useful it'll be in practice to output all the claim records data in this string?

Reply:
only useful for debugging and early testing... thereafter it can be dropped as I'm sure nobody really wants to read a raw byte slice